src/DefinitionResolver.php

```diff
@@ -970,6 +979,11 @@ public function getTypeFromNode($node)
             return $this->resolveExpressionNodeToType($node->argumentExpressionList->children[2]->expression);
         }
 
+        // FOREACH KEY VARIABLE
+        if ($node instanceof Node\ForeachKey) {
+            return new Types\Integer;
```

Comment:
The key of a `foreach` variable doesn't have to be integer. It can be anything. You need to find out by resolving the iterated variable to a type and if it's an array get its `keyType`

Reply:
Does this need special treatment for

`foreach($arr as list($a1, $a2))` 

Like looks?

Reply:
Oops, that was just for testing purposes, and indeed, yes, I think `list($a1, $a2)` needs special treatment, although I do not know how I can achieve that.